### PR TITLE
Create test and common modules

### PR DESCRIPTION
**Pull request by samtstern**
Makes two new modules:

   * `geofire-test` - Contains testing utilities that we intend to publish
   * `geofire-common` - Contains all sources so that modules can share without circular deps

Also removes all use of deprecated APIs.

### CODE REVIEW

**Comment by samtstern** [comment]
This addresses #101

**Comment by samtstern** [comment]
@vanniktech which test classes did you envision exposing in a module?  So far I just moved `TestHelpers`, `GeoFireTestingRule` and `SimpleFuture`

File: java/src/test/java/com/firebase/geofire/GeoLocationTest.java [hunk 1]
```diff
@@ -1,49 +1,54 @@
...
+
+    private static final double EPSILON = 0.0000001;
+
     @Test
     public void geoLocationHasCorrectValues() {
-        Assert.assertEquals(new GeoLocation(1,2).latitude, 1.0);
-        Assert.assertEquals(new GeoLocation(1,2).longitude, 2.0);
-        Assert.assertEquals(new GeoLocation(0.000001,2).latitude, 0.000001);
-        Assert.assertEquals(new GeoLocation(0,0.000001).longitude, 0.000001);
+      assertEquals(new GeoLocation(1,2).latitude, 1.0, EPSILON);
```
Comment (vanniktech) [inline java/src/test/java/com/firebase/geofire/GeoLocationTest.java]: you're using 2 spaces here now

File: test/src/main/java/com/firebase/geofire/GeoFireTestingRule.java [hunk 2]
```diff
@@ -1,48 +1,52 @@
...
     private static final String SERVICE_ACCOUNT_CREDENTIALS = "service-account.json";
 
     private DatabaseReference databaseReference;
 
-    @Override public void starting(Description description) {
+    @Override
+    public void starting(Description description) {
         if (FirebaseApp.getApps().isEmpty()) {
-            final FirebaseCredential credentials;
+            final GoogleCredentials credentials;
```
Comment (vanniktech) [inline test/src/main/java/com/firebase/geofire/GeoFireTestingRule.java]: Nice 👍

**Comment by samtstern** [comment]
A dep from test on the normal module leads to a circular dep error from
maven.

**Comment by vanniktech** [comment]
Huh that's interesting. In Gradle it just works when you use `compile project(':geofire')` from the `test/build.gradle` file.

**Comment by samtstern** [comment]
@vanniktech yeah I was surprised to have it not work but I just can't get it to behave without separating out `common`.  But I don't think it's much of a problem, do you?

And agree, a separate PR to actually document all the test helpers and such.  I just want to lay a foundation.

**Comment by vanniktech** [comment]
> But I don't think it's much of a problem, do you?

Not it should not.

**Comment by vanniktech** [comment]
When executing `mvn -U idea:idea` for generating the idea related files - Otherwise the project does not work from AS in my case - yields `Could not resolve dependencies for project com.firebase:geofire-android:aar:2.2.1-SNAPSHOT: Failure to find com.firebase:geofire-common:jar:2.2.1-SNAPSHOT in file:///Users/nik/android/sdk/extras/google/m2repository was cached in the local repository, resolution will not be reattempted until the update interval of google-extras has elapsed or updates are forced`

**Comment by vanniktech** [comment]
Why is `com.firebase.geofire.example.Example` in the common package? Seems like it's not belonging there.

**Comment by vanniktech** [comment]
- I'd choose a different package name. Maybe `com.firebase.geofire.testing` so it won't conflict (like Database & Database Admin).
- Move `SimpleFuture` out of `util` and into the root
- Package private `TestHelpers` since we don't want this in the public API
- Move the TestingListeners into the testing infrastructure

Let me know how we can proceed. We could make the changes here or later in separate PRs.

**Comment by samtstern** [comment]
@vanniktech I think I addressed all of your comments except for the failure about `mvn -U idea:idea`.  Not quite sure about that one

**Comment by vanniktech** [comment]
@samtstern I haven't forgotten about this. I'll pick this up next week due to some priority changes.

**Pull request merged**